cuda/factorization/par_ilut_filter_kernels.cu

```diff
@@ -12,15 +12,16 @@
 #include <ginkgo/core/matrix/dense.hpp>
 
 
+#include "common/cuda_hip/base/config.hpp"
+#include "common/cuda_hip/base/runtime.hpp"
```

Comment:
when is it necessary to add this header? It's not added to the file above for instance.

Reply:
That's basically an iwyu question - do we include everything directly, or do we use some transitive includes? HIP's runtime is necessary basically any time you call a kernel or refer to some constants like warpSize afaik, I think that cleanup should be part of another PR.